pkg/expr/graph.go

```diff
@@ -143,17 +143,30 @@ func (s *Service) buildGraph(req *Request) (*simple.DirectedGraph, error) {
 			RefID:         query.RefID,
 			TimeRange:     query.TimeRange,
 			QueryType:     query.QueryType,
-			DatasourceUID: query.DatasourceUID,
+			DatasourceUID: query.GetDatasourceUID(),
 		}
 
-		isExpr, err := rn.IsExpressionQuery()
-		if err != nil {
-			return nil, err
+		numericDSID := float64(0) // legacy
+		if rn.DatasourceUID == "" {
```

Comment:
@kylebrandt -- is this path still possible?  is the graph called from anywhere besides `/api/ds/query`?

Reply:
I don't think so, was defaulting to being cautious.

Reply:
Should we remove it now or in a different PR?  I am a bit nervous with the multiple ways to identify a datasource in graph/node but also not clear on where this is called and referenced.